src/googleError.ts

```diff
@@ -23,6 +23,8 @@ export class GoogleError extends Error {
   note?: string;
   metadata?: Metadata;
   statusDetails?: string | protobuf.Message<{}>[];
+  reason?: string;
+  domain?: string;
```

Comment:
If the ErrorInfo is additionally decoded into the statusDetails field, then won't the 'reason' and 'domain' be callable from statusDetails?

Reply:
We treat `ErrorInfo` differently from other error detail type. As one of Error Improvement project, we promote the ErrorInfo as first class field in returning error.
Thanks for reviewing.